src/sage/rings/padics/padic_generic_element.pyx

```diff
@@ -2822,21 +2831,25 @@ cdef class pAdicGenericElement(LocalGenericElement):
             sage: S.<x> = ZZ[]
             sage: K = Qp(5,5)
             sage: f = x^625 - 5*x - 5
+
+            sage: # needs sage.libs.ntl sage.rings.padics
             sage: W.<w> = K.extension(f)
             sage: z = 1 - w^2 + O(w^11)
             sage: x = 1 - z
             sage: z.log().precision_absolute()
             -975
+
```

Comment:
I don't understand why I don't see any warning for `x`. It is created in a block with `# needs...` and used in another block without `# needs...`.

Reply:
`x` is already unconditionally defined, from the global environment, so there won't be warnings

I'm removing the blank line.